stdlib/src/collections/dict.mojo

```diff
@@ -973,6 +986,24 @@ struct Dict[K: KeyElement, V: CollectionElement](
 
         self._n_entries = self.size
 
+    fn _resize_down(inout self):
+        var _new_reserved = self._reserved() >> 1
+        var _previous_list = self._entries^
+        self._entries = __type_of(self._entries)(capacity = _new_reserved)
+    
+        for i in range(_new_reserved): self._entries.append(None)
+        self._index = _DictIndex(_new_reserved)
+        var new_i = 0
+        for i in range(self._n_entries):
+            if _previous_list[i]: 
+                self._set_index(
+                    self._find_empty_index(_previous_list[i].value().hash),
+                    new_i
+                )
+                self._entries[new_i]=_previous_list[i].unsafe_take()
+                new_i += 1
+        self._n_entries = self.size
```

Comment:
Is there enough overlap with `_maybe_resize` to reuse?

Reply:
:+1:  Good idea, yes they are unified in new commits